File: Ruby/season2/p037-043/p043.rb
# -*- coding: utf-8 -*-
=begin
　2017.5.3
　クラスの継承による循環配列定義
=end


# 循環配列（循環バッファ）
class CircularBuffer < Array
  # []演算子の上書き
  def [](i)
    super(i % length)
  end
end


def main
  cb = CircularBuffer["く","ぁ","ｗ","せ","ｄ","ｒ","ｆ","ｔ","ｇ","ｙ","ふ","じ","こ","ｌ","ｐ","；","＠","："]
  p cb
  p cb[0]
  p cb[10]
  p cb[100]
  p cb[-1]
  p cb[-11]
  p cb[-111]
end


main

File: Python/season2/p042-043/p043.py
# coding: UTF-8
"""
　2018.5.23
　モジュールファイルのimport
"""

import json
import sys

rfp = open(sys.argv[1], "r", encoding="utf-8")
wfp = open("all_" + sys.argv[1], "w", encoding="utf-8")

jin = json.load(rfp)
jout = []
for lect in jin:
    if lect["name_j"] == "全教員":
        jout.append(lect)

json.dump(jout, wfp, indent=4, ensure_ascii=False)

rfp.close()
wfp.close()

